mfa/push/src/main/kotlin/com/pingidentity/mfa/push/PushDeviceTokenManager.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.mfa.push
+
+import com.pingidentity.logger.Logger
+import com.pingidentity.mfa.push.storage.PushStorage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ensureActive
+import kotlinx.coroutines.withContext
+import java.util.Date
+
+/**
+ * Manager class for handling push device tokens.
+ * 
+ * This class is responsible for managing device tokens used for push notifications,
+ * including storing, retrieving, and updating tokens locally.
+ *
+ * @property storage The storage implementation for persisting device tokens
+ * @property logger The logger instance
+ */
+class PushDeviceTokenManager(
+    private val storage: PushStorage,
+    private val logger: Logger = Logger.logger
+) {
+    // The current device token
+    private var currentToken: String? = null
+    
+    /**
+     * Get the current push device token
+     * 
+     * @return The current Push device token, or null if not found
+     */
+    suspend fun getCurrentDeviceToken(): PushDeviceToken? = withContext(Dispatchers.IO) {
+        try {
+            return@withContext storage.getCurrentPushDeviceToken()
+        } catch (e: Exception) {
+            logger.e("Failed to get current device token: ${e.message}")
+            return@withContext null
+        }
+    }
+    
+    /**
+     * Get the current device token ID
+     * 
+     * @return The current device token ID as a String, or null if not set
+     */
+    suspend fun getDeviceTokenId(): String? = withContext(Dispatchers.IO) {
```

Comment:
May review overall implementation, wihtContext may not necessary here, should define it to the lowest level of function which actually doing the IO process.